File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        String logMessage = logLine.split(":\s")[1].trim();
        return logMessage;
    }

    public static String logLevel(String logLine) {
        String level = logLine.split(":\s")[0].replace("[", "").replace("]", "").toLowerCase().trim();
        return level;
    }

    public static String reformat(String logLine) {
        String logMessage = message(logLine);
        String level = logLevel(logLine);
        String reformattedLine = logMessage + " (" + level + ")";
        return reformattedLine;
    }
}

File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int indexOfColon = logLine.IndexOf(':');
        return logLine.Substring(indexOfColon + 1).Trim();
    }

    public static string LogLevel(string logLine)
    {
        int indexOfBeginning = logLine.IndexOf('[');
        int indexOfEnd = logLine.IndexOf(']');

        return logLine.Substring(indexOfBeginning + 1, indexOfEnd - indexOfBeginning - 1).ToLower();
    }

    public static string Reformat(string logLine)
    {
        string logLevel = LogLevel(logLine);
        string message = Message(logLine);

        return $"{message} ({logLevel})";

    }
}
